### ISSUE TITLE
[Bug]: Heatmaps are different when using original map and torch inference in efficientAD

### ISSUE THREAD

**Issue by 840691168** [open-edge-platform/anomalib#2726]
### Describe the bug

I am trying to use torch infer the same image. However, the heatmaps are varied and cause many false detection in the prediction mask.(my own datasets)
This is the torch_heatmap

![Image](https://github.com/user-attachments/assets/ea9ca6cd-ee00-4829-b3d1-6de4cef1a443)

while the ori_heatmap is 

![Image](https://github.com/user-attachments/assets/e1a00772-5c7f-4b26-ae22-65799514749f)

By the way ,I have already resize the input picture to (256,256) but the heatmaps are same.

### Dataset

Other (please specify in the text field below)

### Model

Other (please specify in the field below)

### Steps to reproduce the behavior

This is my inference code. 
def save_anomaly_heatmap_only(predictions, save_path="heatmap_only.png", threshold=0, overlay_mode=False):
    anomaly_map = predictions.anomaly_map[0].detach().cpu().numpy()
    heatmap_gray = (anomaly_map * 255).astype("uint8")
    heatmap_color = cv2.applyColorMap(heatmap_gray, cv2.COLORMAP_JET)

    if overlay_mode:
        original = predictions.image[0].permute(1, 2, 0).detach().cpu().numpy()
        original = (original * 255).astype("uint8")

        heatmap_resized = cv2.resize(heatmap_color, (original.shape[1], original.shape[0]), interpolation=cv2.INTER_LINEAR)
        heatmap_mask = cv2.resize(anomaly_map, (original.shape[1], original.shape[0]))

        mask = heatmap_mask > threshold
        overlay = original.copy()
        overlay[mask] = (original[mask] * 0.6 + heatmap_resized[mask] * 0.4).astype("uint8")

        cv2.imwrite(save_path, overlay)
        print(f"[✓] ：{save_path}")
    else:
        heatmap_color_resized = cv2.resize(heatmap_gray, (256, 256), interpolation=cv2.INTER_LINEAR)  
        cv2.imwrite(save_path, heatmap_color_resized)
        print(f"[✓] ：{save_path}")

model = TorchInferencer(path="results/weights/torch/model.pt")

input_dir = "datasets/CA_CX/test/good"
output_dir = "heatmaps"
os.makedirs(output_dir, exist_ok=True)

overlay_mode = True  

for filename in os.listdir(input_dir):
    if filename.lower().endswith((".jpg", ".jpeg", ".png", ".bmp")):
        image_path = os.path.join(input_dir, filename)
        image = read_image(image_path)

        predictions = model.predict(image=image)

        output_path = os.path.join(output_dir, f"{os.path.splitext(filename)[0]}_heatmap.png")
        save_anomaly_heatmap_only(predictions, output_path, overlay_mode=overlay_mode)

### OS information

OS information:
- OS: [e.g. Ubuntu 20.04]
- Python version: [e.g. 3.10.0]
- Anomalib version: [e.g. 0.3.6]
- PyTorch version: [e.g. 1.9.0]
- CUDA/cuDNN version: [e.g. 11.1]
- GPU models and configuration: [e.g. 2x GeForce RTX 3090]
- Any other relevant information: [e.g. I'm using a custom dataset]

### Expected behavior

Make the heatmaps identical when using different inference model.

### Pip/GitHub

GitHub

### Configuration YAML

```yaml
No yaml
```

### Logs

```shell
no log
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by samet-akcay** [comment]
@840691168, would it be related to your dataset? Do these tests images significantly deviate from your validation images? 

I'm trying to understand if this is a bug or a dataset-related problem.

**Comment by 840691168** [comment]
> [@840691168](https://github.com/840691168), would it be related to your dataset? Do these tests images significantly deviate from your validation images?
> 
> I'm trying to understand if this is a bug or a dataset-related problem.

@samet-akcay  

Hi Samet,

Thanks for your follow-up. I’m really sorry that I can’t directly share my training images. However, I can describe my use case: it’s for detecting whether components on a PCB are properly soldered and complete. All the items being inspected are of the same model, and the camera misalignment between images is less than 20 pixels.

![Image](https://github.com/user-attachments/assets/04569480-b465-4393-9d4c-70b3ac598edf)

Even when defects occur (e.g., one or two missing screws), the overall image structure remains very similar. So I can confidently say that the differences between test and validation images are not the issue here.

What’s noteworthy is that I used the same checkpoint file, and ran inference on the same image using three different methods:

engine.test

Torch model with batch input

ONNX inference

Interestingly, Torch and ONNX give consistent results, but engine.test produces noticeably different heatmaps.

Hope this helps narrow things down. Let me know if you need any more details!

**Comment by samet-akcay** [comment]
I think I've found the root cause and created a dedicated issue. This one can be closed, and the updates could be followed from [🐞 Discrepancy in Inference Results Between Lightning, Torch, ONNX, and OpenVINO Models](https://github.com/open-edge-platform/anomalib/issues/2747)
